install/kubernetes/helm/istio/charts/prometheus/templates/deployment.yaml

```diff
@@ -76,6 +82,13 @@ spec:
           defaultMode: 420
           optional: true
           secretName: istio.default
+      - name: storage-volume
+        {{- if .Values.persistentVolume.enabled }}
+          persistentVolumeClaim:
+            claimName: {{ if .Values.persistentVolume.existingClaim }}{{ .Values.persistentVolume.existingClaim }}{{- else }}{{ template "prometheus.fullname" . }}{{- end }}
+        {{- else }}
+          emptyDir: {}
+        {{- end -}}
```

Comment:
you have a wrong yaml syntax here. 
persistentVolumeClaim and emptyDir  should be on the same level as name:
{{- end -}} should be {{- end }} otherwise you wont get a needed newline: 

`      - name: storage-volume
        {{- if .Values.persistentVolume.enabled }}
        persistentVolumeClaim:
          claimName: {{ if .Values.persistentVolume.existingClaim }}{{ .Values.persistentVolume.existingClaim }}{{- else }}{{ template "prometheus.fullname" . }}{{- end }}
        {{- else }}
        emptyDir: {}
        {{- end }}
`

Reply:
Updated accordingly.

Reply:
I created a PR for your fork also. Data folder has wrong permissions by default. I added init cointainer to fix it.

Reply:
Thanks. It has been merged.